ui/app/AppLayouts/Onboarding2/OnboardingFlow.qml

```diff
@@ -201,7 +200,14 @@ OnboardingStackView {
             }
             onDismissBiometricsRequested: root.dismissBiometricsRequested()
             onLoginRequested: (keyUid, method, data) => root.loginRequested(keyUid, method, data)
-            onOnboardingCreateProfileFlowRequested: root.push(createProfilePage)
+            onOnboardingCreateProfileFlowRequested: {
+                const page = root.push(helpUsImproveStatusPage)
+
+                page.shareUsageDataRequested.connect(enabled => {
+                    root.shareUsageDataRequested(enabled)
+                    root.replace(createProfilePage)
+                })
```

Comment:
This connect code is the same as above. Can't you just move this to the `onShareUsageDataRequested` of `HelpUsImproveStatusPage` since both use that same component?

Reply:
But the logic is different in either case, one path loads a specified `component`, in the other we always load the create profile page

Reply:
Ah yes my bad